src/Umbraco.Web.UI.Client/src/common/services/assets.service.js

```diff
@@ -89,12 +95,13 @@ angular.module('umbraco.services')
         }
 
         /**
-         * Loads specific Moment.js Locales.
+         * Loads specific Moment.js and Flatpickr Locales.
          * @param {any} locales
          * @param {any} supportedLocales
          */
         function loadLocales(locales, supportedLocales) {
-            var localeUrls = getMomentLocales(locales, supportedLocales);
+            var localeUrls = getMomentLocales(locales, supportedLocales.moment);
+            localeUrls = localeUrls.concat(getFlatpickrLocales(locales, supportedLocales.flatpickr));
             if (localeUrls.length >= 1) {
```

Comment:
It seems to load both `momentjs` and `flatpickr` locales. Maybe it should be possible to load each of them? E.g. only load `flatpickr` locales where needed in `umb-flat-pickr` directive.

In v7 it seems `loadMomentLocaleForCurrentUser` (changed to loadLocaleForCurrentUser here) only is used here.
https://github.com/umbraco/Umbraco-CMS/blob/dd6e764588d22ef2b7bce01fd504ece89834f181/src/Umbraco.Web.UI.Client/src/init.js#L23

Reply:
@bjarnef they're both loaded on load, so no matter your approach they both need to be loaded at roughly the same time. 

I did start out with an implementation that split the loading into two from a principal perspective of single responsibility. But given that this is only called in that one place, it seems more pragmatic to have one request that plows through the server disk, looking for JS files, instead of two of them.